packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -387,41 +387,35 @@ const Dashboard: FC = () => {
         useState<string>();
 
     useEffect(() => {
-        const checkReload = (event: BeforeUnloadEvent) => {
-            if (isEditMode && (haveTilesChanged || haveFiltersChanged)) {
-                const message =
-                    'You have unsaved changes to your dashboard! Are you sure you want to leave without saving?';
-                event.returnValue = message;
-                return message;
-            }
-        };
-        window.addEventListener('beforeunload', checkReload);
-        return () => window.removeEventListener('beforeunload', checkReload);
-    }, [haveTilesChanged, haveFiltersChanged, isEditMode]);
+        // Check if in edit mode and changes have been made
+        if (isEditMode && (haveTilesChanged || haveFiltersChanged)) {
+            // Define the navigation block function
+            const navigationBlockFunction = (prompt: { pathname: string }) => {
+                // Check if the user is navigating away from the current dashboard
+                if (
+                    !prompt.pathname.includes(
+                        `/projects/${projectUuid}/dashboards/${dashboardUuid}`,
+                    )
+                ) {
+                    // Set the blocked navigation location to navigate on confirming from user
+                    setBlockedNavigationLocation(prompt.pathname);
+                    // Open a warning modal before blocking navigation
+                    setIsSaveWarningModalOpen(true);
+                    // Return false to block history navigation
+                    return false;
+                }
+                // Allow history navigation
+                return undefined;
+            };
 
-    useEffect(() => {
-        const createChartInDashboardFlow = sessionStorage.getItem(
-            'unsavedDashboardTiles',
-        );
-        history.block((prompt) => {
-            if (
-                isEditMode &&
-                (haveTilesChanged || haveFiltersChanged) &&
-                !prompt.pathname.includes(
-                    `/projects/${projectUuid}/dashboards/${dashboardUuid}`,
-                ) &&
-                createChartInDashboardFlow
```

Comment:
@ZeRego do you remember if we are supposed to block navigation if someone is trying to add a chart in dashboard ? I think this is what the condition was for 

[Screencast from 18-09-23 09:25:39.webm](https://github.com/lightdash/lightdash/assets/1983672/f136973c-afdb-4925-8e68-720f3576d6d9)

Reply:
This wasn't really handled well before. Fixing it might be what caused the reload warning to break to begin with. With the current design we should *not* show this warning when the user adds a chart to a dashboard.

Reply:
Made the change to allow user to add a chart to the dashboard without a warning